packages/xstate-solid/src/fsm.ts

```diff
@@ -0,0 +1,101 @@
+import type {
+  EventObject,
+  MachineImplementationsFrom,
+  StateMachine,
+  StateFrom,
+  ServiceFrom,
+  Typestate
+} from '@xstate/fsm';
+import { interpret, createMachine } from '@xstate/fsm';
+
+import { createStore, reconcile } from 'solid-js/store';
+import type { Accessor } from 'solid-js';
+
+import { $PROXY, batch, createEffect, on, onCleanup, onMount } from 'solid-js';
+
+const getServiceState = <
+  TContext extends object,
+  TEvent extends EventObject = EventObject,
+  TState extends Typestate<TContext> = { value: any; context: TContext }
+>(
+  service: StateMachine.Service<TContext, TEvent, TState>
+): StateMachine.State<TContext, TEvent, TState> => {
+  let currentValue: StateMachine.State<TContext, TEvent, TState>;
+  service
+    .subscribe((state) => {
+      currentValue = state;
+    })
+    .unsubscribe();
+  return currentValue!;
+};
+
+export function useMachine<TMachine extends StateMachine.AnyMachine>(
```

Comment:
Shouldn't this reuse `createService` in its implementation? as it doesn't do that it doesn't currently handle `options.state` at all - while `createService` handles it